### ISSUE TITLE
Touch drag on canvas activates navigation controller?

### ISSUE THREAD

**Issue by danomatika** [danomatika/PdParty#88]
Reported by @Josef-N:

>...sometimes, when moving around the screen (on iPhone 6s), suddenly the page turns from the active scene to the last used page, whatever is displayed in the upper left corner of the title bar – e.g.:
< Documents. Once it happened, while moving around the screen, the page turned to documents, and I deleted a patch, before I could release my finger.
Do you see any possibility to make the screen safer?
But I think, this is also a situation, that will not happen very frequently.

**Comment by danomatika** [comment]
Can you share a minimal version of the patch or a patch with similar behavior for testing?

**Comment by danomatika** [comment]
Can you share a minimal version of the patch or a patch with similar behavior for testing?

**Comment by danomatika** [comment]
Can you provide the exact steps to cause this? Does it happen only with a specific patch?

**Comment by Josef-N** [comment]
yes, only with a specific patch. But on the other hand, I did not use PdPart in this way before, with big and fast movements on screen. For this issue it is difficult to make a minimal version, I don’t know where to start. I send the complete zipped scene-folder.
It does not happen if only playing notes, i.e. using touch_down and touch_up, of course. Only with touch_xy, and more likely with two ore more voices. In my patch for instance, when radiobutton "Touch" or toggle "Gliss" is enabled.
I am not absolutely sure, but I think it happens when moving diagonally.
And it happens only every now and then. I hope, this helps a little bit.

[Touch-Keys.zip](https://github.com/danomatika/PdParty/files/5898175/Touch-Keys.zip)

**Comment by Josef-N** [comment]
I tried again for a longer time now, playing around with this patch, but this issue doesn’t occur when I want it to. So I think, it is impossible for you to test, if I cannot provide exact information.
I would suggest, that we forget it for now, and I will write again, if I find out, what exactly causes that. It happens really very rarely anyway. Sorry for that!

**Comment by Josef-N** [comment]
Finally I found out what is happening: it is certainly not a bug.
Since I could not replicate it any more, I watched my fingers, what they are doing. And I observed, that in some situations one of my fingers is very close to this navigation controller, and I am quite sure now, that I touched it accidentally.
So from my side there is no need to change anything in PdParty related to this.
One solution I could imagine for making the screen absolutely save is described here:
hide/show nav bar & status bar #29
This would be nice anyway, but I guess, it is not on top of your to-do list.
Now that I know what causes it, I will take care of my fingers, and for me the problem is solved. It is just natural, that things can happen on a very small display.

**Comment by Josef-N** [comment]
One more consideration:
the only critical situation is, I think, if the current navigation controller points to "< Browser". In worst case, what happened to me, the continued touch-dragging (to the left) results in swiping out a file in the Documents list, and so deleting it without confirmation.
Would it be possible (and not tooo much work), that the user has to confirm the deletion after touching the red "Delete" button?
This might be good anyway, and the perfect solution in my opinion.

**Comment by danomatika** [comment]
The swipe to delete is the default table behavior and there isn't anything customized. It takes two actions: 1. swipe left on the table time, 2. tap on the red delete button.

>This might be good anyway, and the perfect solution in my opinion.

That would turn 2 actions into 3 which could become an issue of annoyance to probably many.

Do you have Multitouch Gestures enabled in the settings? These can conflict with multitouch track within an app. It would also help explain why a left moving gesture somehow translates over to a continuing leftward gesture & tap after the segue to the previous navigation layer.

**Comment by Josef-N** [comment]
> That would turn 2 actions into 3 which could become an issue of annoyance to probably many.

Ok, I understand, this is a good argument.
On iPad I have gestures disabled, but I am using touch in this way only on iPhone, and there I cannot find these settings.
What I wrote about left moving was not correct, sorry. My layout is landscape, the phone rotated by 180 degrees, with the lightning connector on the right side. What I tried to reproduce:
Tapping on "< Browser" or "< Documents" while touch-dragging on screen with another finger. Of course I try to avoid that, but sometimes it happens accidentally. Then it depends on the orientation of the device. In my case I am moving downwards. When the page turns to Documents, the phone changes to portrait view, and the previous down-movement changes to / continues leftwards.
The rest is only an assumption, what could have happened, as I never could reproduce deleting something.
Regarding your update (delay of 100 or 200 ms):
There have been several updates to iOS since I first reported, I think the whole thing is obsolete now, BECAUSE:
In the meantime I compared all apps that I am frequently using, and my conclusion is, that PdParty has a very good mixture of things that are allowed or deferred in communication with the operating system.
